src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -277,7 +283,15 @@ ReactDOMComponent.Mixin = {
   mountComponent: function(rootID, transaction, context) {
     this._rootNodeID = rootID;
 
-    assertValidProps(this, this._currentElement.props);
+    var props = this._currentElement.props;
+
+    var wrapper = wrappers[this._tag];
+    if (wrapper) {
+      wrapper.mountWrapper.call(this, props, context);
```

Comment:
It does parse the code, though I think it supports ES6 fine (or at least claims to). Pretty sure you can run sweet.js then babel, though of course this would be more tooling and could hinder adoption of things like Flow later.